Neos.Eel/Classes/Helper/StringHelper.php

```diff
@@ -24,6 +24,43 @@
  */
 class StringHelper implements ProtectedContextAwareInterface
 {
+    /**
+     * Generates a slug of the given string
+     *
+     * Examples::
+     *
+     *     String.urlize('Hello World') == 'hello-world'
+     *     String.urlize('Ä_ÖÜßäöüÅåAÆæØEuro:€éñåøå!_И люяюлб PHP! есть. ﬁ  ') == 'ae-oeuessaeoeueaaaaaaeaeoeuroeenaaoaa-i-luaulb-php-est-fi'
+     *
+     * @param string $string The string to be transformed.
+     * @return string The converted string
+     */
+
+    public function urlize(string $string): string
+    {
+        $specialChars = [
```

Comment:
Why do we do this conversion here and then transliterate, that seems wrong? Shouldn't the Transliteration take care of these?

Reply:
True

Reply:
Nope. The transliteration (with this config) doesn't take care of these chars. But as I said: If we decide not to implement behat, I will further dive deeper into this and optimize the code 😉

Reply:
Wait, what do you mean it doesn't take care of that? It doesn't transliterate umlauts to something else?

Reply:
I think the example config I posted didn't cover special characters like "€". They could all be added (see http://userguide.icu-project.org/transforms/general).
But the question still remains _what_ rules to apply.. Maybe there's some official mapping for all disallowed URL path characters

Reply:
Weeell, that's the issue with naming, right. If you want to build a URL with special characters I would argue you urlencode instead of doing some transliteration ;) So whatever is allowed or not there doesn't _really_ matter. This is more of a asciilize I guess? IDK. The problem with transliteration is that there is no universal solution because you loose information. The only way to retain at least part of the information of the original string would be to know it's langauge because the "manual" mapping we do there might be right "for us" but not necessarily for someone else. So either this needs to be given a source language to transliterate for OR must follow basic asciifolding rules, aka ü => u, ú => u etc.

Reply:
True that. "disallowed" wasn't the right word.
But there gotta be some standards that define mappings like `ü` => `ue` and similar.. I hoped..

Reply:
Just to quote from some other text:

> In German the ¨ (umlaut) is always transliterated as `-e`, while in Swedish the diacritic is simply dropped.

There is no universal rule, most diacritics are transliterated differently depending on their source language, so unless you give that only a "flattening" transliteration is sensible.

Reply:
Yes, that's a good call.. It _could_ mean that a helper expects an (optional?) locale or preset but maybe that takes it too far